src/options.ts

```diff
@@ -6,37 +26,44 @@ const TypedArray = Object.getPrototypeOf(Uint8Array);
 const objectToString = Object.prototype.toString;
 
 // This allows transforms to behave equivalently to channels.
-export function valueof(data, value, arrayType) {
-  const type = typeof value;
-  return type === "string"
+/**
+ * @link https://github.com/observablehq/plot/blob/main/README.md#plotvalueofdata-value-type
+ */
+export function valueof(
+  data: Data,
+  value: ChannelOption | ChannelTransformAccessor | null | undefined,
+  arrayType: ArrayType
+) {
+  return typeof value === "string"
     ? map(data, field(value), arrayType)
-    : type === "function"
+    : typeof value === "function"
     ? map(data, value, arrayType)
-    : type === "number" || value instanceof Date || type === "boolean"
+    : typeof value === "number" || value instanceof Date || typeof value === "boolean"
```

Comment:
I'm seeing warnings where the union of the type is under specified.  
```
Warning:(41, 7) Invalid 'typeof' check: 'value' cannot have type 'number'
Warning:(41, 36) Invalid 'instanceof' check: 'value' has type that is not related to 'Date'
Warning:(41, 61) Invalid 'typeof' check: 'value' cannot have type 'boolean'
```
A naive fix.. just append said types to the args list. e.g. `value: ... | number | Date | boolean,`.

Or maybe instead they should be added to the already defined types, e.g. `ChannelOption`, if not too broad 🤷 .

Reply:
Thanks for flagging! I don't get these warnings myself, not in VSCode nor in `yarn test:typecheck`. How come?

Reply:
Are you using the --watch option?